src/hb-array.hh

```diff
@@ -171,6 +171,21 @@ struct hb_array_t : hb_iter_with_fallback_t<hb_array_t<Type>, Type&>
 
   unsigned int get_size () const { return length * this->get_item_size (); }
 
+  void reverse ()
+  {
+    int rhs = length - 1;
+    int lhs = 0;
+    while (rhs > lhs)
+    {
+      Type value_rhs = arrayZ[rhs];
+      Type value_lhs = arrayZ[lhs];
```

Comment:
This uses two temps. Please do with just one. Make buffer's reverse call this.  Add start/end so that works.